library/core-utils/src/main/java/im/vector/lib/core/utils/compat/MutableCollectionCompat.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.lib.core.utils.compat
+
+import android.os.Build
+
+fun <E> MutableCollection<E>.removeIfCompat(predicate: (E) -> Boolean) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        removeIf(predicate)
+    } else {
+        removeAll(filter(predicate).toSet())
```

Comment:
I have been asked to convert to Set for performance reason.

Reply:
as this is only for the _keys to remove_ there shouldn't be any changes for lists with duplicated items?

```kotlin
listOf(1, 1, 2).removeIfCompat { it == 2 }  // would still expect [1, 1] 
```

Reply:
Now that you say it, I am not sure what will happen with
`listOf(1, 1, 2).removeIfCompat { it == 1 }  // would still expect [2] but I think we will get [1, 2]`
Let me play a bit with it.

Reply:
Seems to work as expected:
<img width="904" alt="image" src="https://user-images.githubusercontent.com/3940906/149775547-f9fe8eb2-3273-4359-b0e8-d99609d3a3ed.png">
(https://pl.kotl.in/vIVLDMXe-)

Reply:
thanks for checking!